python/node_similarity.py

```diff
@@ -0,0 +1,289 @@
+        node2: mgp.Any,
+        mode: str = 'c'
+) -> mgp.Record(node1=mgp.Vertex, node2=mgp.Vertex, similarity=mgp.Number):
+    """
+    This procedure calls the method for calculating the overlap similarity between two nodes (or two lists of nodes)
+    and returns 3 fields
+
+    :return node1: The first node
+    :return node2: The second node
+    :return similarity: The overlap similarity of the first and the second node
+
+    The input arguments consist of
+
+    :param mgp.Vertex or Tuple[mgp.Vertex] node1: A node or a tuple of nodes
+    :param mgp.Vertex or Tuple[mgp.Vertex] node2: A node or a tuple od nodes
+    :param str mode: Can be `p` for pairwise similarity calculation or `c` for calculating the similarity of the
+    Cartesian product of given tuples. Default value is `c`.
+    """
+
+    nodes1: tuple
+    nodes2: tuple
+
+    if isinstance(node1, mgp.Vertex):
+        nodes1 = tuple([node1])
+    elif isinstance(node1, tuple):
+        nodes1 = node1
+    else:
+        raise TypeError("Invalid type of first argument.")
+
+    if isinstance(node2, mgp.Vertex):
+        nodes2 = tuple([node2])
+    elif isinstance(node2, tuple):
+        nodes2 = node2
+    else:
+        raise TypeError("Invalid type of second argument.")
+
+    result = _calculate_similarity(nodes1, nodes2, _calculate_overlap, mode)
+
+    return [mgp.Record(node1=n1, node2=n2, similarity=similarity) for n1, n2, similarity in result]
+
+
+@mgp.read_proc
+def cosine(
+        context: mgp.ProcCtx,
+        node1: mgp.Any,
+        node2: mgp.Any,
+        mode: str = 'c'
+) -> mgp.Record(node1=mgp.Vertex, node2=mgp.Vertex, similarity=mgp.Number):
+    """
+    This procedure calls the method for calculating the cosine similarity between two nodes (or two lists of nodes)
+    and returns 3 fields
+
+    :return node1: The first node
+    :return node2: The second node
+    :return similarity: The cosine similarity of the first and the second node
+
+    The input arguments consist of
+
+    :param mgp.Vertex or Tuple[mgp.Vertex] node1: A node or a tuple of nodes
+    :param mgp.Vertex or Tuple[mgp.Vertex] node2: A node or a tuple od nodes
+    :param str mode: Can be `p` for pairwise similarity calculation or `c` for calculating the similarity of the
+    Cartesian product of given tuples. Default value is `c`.
+    """
+
+    nodes1: tuple
+    nodes2: tuple
+
+    if isinstance(node1, mgp.Vertex):
+        nodes1 = tuple([node1])
+    elif isinstance(node1, tuple):
+        nodes1 = node1
+    else:
+        raise TypeError("Invalid type of first argument.")
+
+    if isinstance(node2, mgp.Vertex):
+        nodes2 = tuple([node2])
+    elif isinstance(node2, tuple):
+        nodes2 = node2
+    else:
+        raise TypeError("Invalid type of second argument.")
+
+    result = _calculate_similarity(nodes1, nodes2, _calculate_cosine, mode)
+
+    return [mgp.Record(node1=n1, node2=n2, similarity=similarity) for n1, n2, similarity in result]
```

Comment:
Since this is pretty repetitive, maybe it would be better to export it in a single function, where the only thing that differs is method of choice for calculation